From dd76e415b2a707d31a125201c5d21aadb07282a7 Mon Sep 17 00:00:00 2001
From: James Le Cuirot <chewi@gentoo.org>
Date: Tue, 11 Jun 2024 15:39:25 +0100
Subject: [PATCH] ebuild.profiles: Fix case where a parent path omits the repo
 identifier

According "SPECIFIC FILE DESCRIPTIONS" in `man portage`, it is valid to
have a parent path like `:path/to/profile` where the repo identifier is
missing. This refers to a path in the current repo.

Signed-off-by: James Le Cuirot <chewi@gentoo.org>
---
 src/pkgcore/ebuild/profiles.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/pkgcore/ebuild/profiles.py b/src/pkgcore/ebuild/profiles.py
index 384e41e22..fea1eacc7 100644
--- a/src/pkgcore/ebuild/profiles.py
+++ b/src/pkgcore/ebuild/profiles.py
@@ -254,6 +254,8 @@ def parent_paths(self, data):
                                     f"unknown repo {repo_id!r}"
                                 )
                                 continue
+                    else:
+                        location = repo_config.location
                     l.append(
                         (
                             abspath(pjoin(location, "profiles", profile_path)),
